File: ReadyChef.Infrastructure/Services/RecipeService.cs
using ReadyChef.Core.Services;
using System;
using System.Collections.Generic;
using ReadyChef.Core.Models;
using ReadyChef.Core.Repositories;

namespace ReadyChef.Infrastructure.Services
{
    public class RecipeService : IRecipeService
    {
	    private readonly IRecipeRepository _recipeRepository;

	    public RecipeService(IRecipeRepository recipeRepository)
	    {
		    _recipeRepository = recipeRepository;
	    }


	    public IEnumerable<Recipe> GetAllRecipes()
        {
            return _recipeRepository.GetAll();
        }

        public IEnumerable<Recipe> GetAllRecipes(string ingredient)
        {
            throw new NotImplementedException();
        }

        public Recipe GetRecipe(string name)
        {
            return _recipeRepository.Get(name);
        }

        public void AddRecipe(Recipe recipe)
        {
            _recipeRepository.Add(recipe);
        }

        public IEnumerable<Recipe> GetByName(string name)
        {
            return _recipeRepository.GetByName(name);
        }

        public IEnumerable<Recipe> GetByIngredient(string ingredient)
        {
            return _recipeRepository.GetByIngredient(ingredient);
        }
    }
}

File: ReadyChef.Api/Scripts/services/recipeService.js
angular.module('readyChefApp')
    .factory('recipeService',recipeService);
recipeService.$inject=['recipe','$http'];
function recipeService(recipe, $http) {
    var vm = this;
    

    function saveRecipe(recipe) {
        $http.post('/api/Recipe/Add', recipe);
    }

    function getRecipesByName(name) {
        return $http.get('/api/Recipe/GetByName/' + name)
                .then(function (response) {
                    return response.data;
                })
                .catch(function (error) {
                    console.log(error);
                });
    }

    function getRecipesByIngredient(ingredient) {
        return $http.get('/api/Recipe/GetByIngredient/' + ingredient)
                .then(function (response) {
                    return response.data;
                })
                .catch(function (error) {
                    console.log(error);
                });
    }

    function getAllRecipes() {
        return $http.get('/api/Recipe/getall')
                .then(function (response) {
                    return response.data;
                })
                .catch(function (error) {
                    console.log(error);
                });
    };

    return {
        saveRecipe: saveRecipe,
        getRecipesByName: getRecipesByName,
        getRecipesByIngredient: getRecipesByIngredient,
        getAllRecipes: getAllRecipes
    };
};